### PR TITLE
adjust check for smtp without auth

### LINKED ISSUE #1066: Configuring Greenlight to send notifications via local SMTP server that does not require authentication
Hello,
I have just set up a new BigBlueButton server and installed Greenlight, now I'm trying to configure it to send notifications via our local, internal SMTP server, which does not require (or support) authentication. No matter what I put for "SMTP_AUTH", I always get:
```
Checking SMTP connection: Failed
Error connecting to SMTP - wrong authentication type
```
Since there is no documentation for the SMTP settings, I don't know which value to put for "SMTP_AUTH" to say **none**.

### PR DESCRIPTION
**Pull request by einhirn**
this should do it. 
Cleaned up commit history after forking this repo anew.
Supersedes #1125
fixes #1066

### CODE REVIEW

File: lib/tasks/configuration.rake [hunk 1]
```diff
@@ -44,8 +44,10 @@ def test_smtp
     smtp.enable_starttls_auto if smtp.respond_to?(:enable_starttls_auto)
   end
 
-  smtp.start(ENV['SMTP_DOMAIN'], ENV['SMTP_USERNAME'], ENV['SMTP_PASSWORD'],
-    ENV['SMTP_AUTH']) do |s|
+  user = ENV['SMTP_USERNAME'].present? ? ENV['SMTP_USERNAME'] : nil
```
Comment (farhatahmad) [inline lib/tasks/configuration.rake]: Can be cleaned up with `user = ENV['SMTP_USERNAME'].presence || nil`

File: lib/tasks/configuration.rake [hunk 2]
```diff
@@ -44,8 +44,10 @@ def test_smtp
...
+  password = ENV['SMTP_PASSWORD'].present? ? ENV['SMTP_PASSWORD'] : nil
```
Comment (farhatahmad) [inline lib/tasks/configuration.rake]: Same as above

**Comment by farhatahmad** [comment]
@einhirn Thank you for taking the time to fix it. I recommended some tiny syntax changes. Once done we'll merge this into our next release

**Comment by einhirn** [comment]
While I'm at it - should this do the same as before?
` authtype = ENV['SMTP_AUTH'].presence && ENV['SMTP_AUTH'] != "none" || nil`

**Comment by einhirn** [comment]
Thinking about it, it probably won't so I won't change that line...

**Comment by jfederico** [comment]
Can you guys rebase this PR?

**Comment by einhirn** [comment]
Hrm... Somehow I always break my repo trying to do that:
```
# git fetch upstream
From https://github.com/bigbluebutton/greenlight
....
# git rebase upstream/master
[looks ok]
# git push
To github.com:rztuc/greenlight.git
 ! [rejected]        adjust-check-for-smtp-without-auth -> adjust-check-for-smtp-without-auth (non-fast-forward)
error: failed to push some refs to 'git@github.com:rztuc/greenlight.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

**Comment by farhatahmad** [comment]
@einhirn Do a rebase and a force push, not a merge

**Comment by farhatahmad** [comment]
See #1155

**Pull request closed without merging**